src/JWTAuthMiddleware.ts

```diff
@@ -188,7 +190,7 @@ export default class JWTAuthMiddleware {
         } as User;
 
         newUser = Object.assign(newUser, await this.beforeUserCreate(newUser, jwtPayload));
-        const user = await saveUpsertWithWhere(this.user, {email}, newUser) as User;
+        const user = await saveUpsertWithWhere(this.user, { id: userId }, newUser) as User;
```

Comment:
The lib should be able to work without the id identifier option. 
Maybe we should switch if the id is not null use the id otherwise use the email?